recipes/catch2/2.x.x/test_package/conanfile.py

```diff
@@ -1,24 +1,43 @@
-from conans import ConanFile, CMake, tools
-from conans.tools import Version
+from conan import ConanFile
+from conan.tools.cmake import CMake, CMakeToolchain
+from conan.tools.build import can_run
+from conan.tools.cmake import cmake_layout
 import os
 
-
 class TestPackageConan(ConanFile):
     settings = "os", "compiler", "build_type", "arch"
-    generators = "cmake", "cmake_find_package"
+    generators = "CMakeDeps", "VirtualRunEnv"
+    test_type = "explicit"
+
+    _tests_todo = []
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def generate(self):
+        tc = CMakeToolchain(self)
+        catch_opts = self.dependencies[self.tested_reference_str].options
+        tc.variables["WITH_PREFIX"]    = catch_opts.with_prefix
+        tc.variables["WITH_MAIN"]      = catch_opts.with_main
+        tc.variables["WITH_BENCHMARK"] = not catch_opts.with_prefix and catch_opts.with_main and catch_opts.with_benchmark
+        tc.generate()
+
+        # note: this is required as self.dependencies is not available in test()
+        self._tests_todo.append("test_package")
+        if catch_opts.with_main:
+            self._tests_todo.append("standalone")
+        if not catch_opts.with_prefix and catch_opts.with_main and catch_opts.with_benchmark:
+            self._tests_todo.append("benchmark")
```

Comment:
```suggestion
        # note: this is required as self.dependencies is not available in test()
        self._tests_todo.append("test_package")
        if catch_opts.with_main:
            self._tests_todo.append("standalone")
        if not catch_opts.with_prefix and catch_opts.with_main and catch_opts.with_benchmark:
            self._tests_todo.append("benchmark")
```

It does not work. Conan parse twice the recipe folder and _tests_todo will be empty as soon as test() is called.

You can create a temporary file and and those values. More information: https://github.com/conan-io/conan/issues/12411

Reply:
I did this in https://github.com/conan-io/conan-center-index/pull/13794/commits/332eb4054b2821be65ad142cc341b037f953da4f

Reply:
I cherry-picked your commit and fixed it up slightly